### ISSUE TITLE
QBEC can't update CustomResource with validation on v1.15

### ISSUE THREAD

**Issue by kvaps** [splunk/qbec#51]
## Bug report

**Environment**

Kubernetes:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.1", GitCommit:"4485c6f18cee9a5d3c3b4e523bd27972b1b53892", GitTreeState:"clean", BuildDate:"2019-07-18T09:18:22Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.1", GitCommit:"4485c6f18cee9a5d3c3b4e523bd27972b1b53892", GitTreeState:"clean", BuildDate:"2019-07-18T09:09:21Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
Qbec:
```
qbec version: 0.7.2
jsonnet version: v0.13.0
go version: 1.12.5
commit: ba28218
```

**Steps to reproduce:**
```
cd /tmp
qbec init memcached-operator
cd memcached-operator

cat > components/crd.jsonnet <<EOT
{
  apiVersion: 'apiextensions.k8s.io/v1beta1',
  kind: 'CustomResourceDefinition',
  metadata: { name: 'memcacheds.cache.example.com' },
  spec: {
    group: 'cache.example.com',
    names: {
      kind: 'Memcached',
      listKind: 'MemcachedList',
      plural: 'memcacheds',
      singular: 'memcached',
    },
    scope: 'Namespaced',
    subresources: { status: {} },
    validation: {
      openAPIV3Schema: {
        properties: {
          spec: {
            properties: {
              configure: { type: 'boolean' },
              size: {
                type: 'integer',
              },
            },
            required: ['size'],
            type: 'object',
          },
        },
        type: 'object',
      },
    },
    version: 'v1beta1',
    versions: [
      {
        name: 'v1beta1',
        served: true,
        storage: true,
      },
    ],
  },
}
EOT

cat > components/cr.jsonnet <<EOT
[
{
  apiVersion: "cache.example.com/v1beta1",
  kind: "Memcached",
  metadata: { name: "example-memcached" },
  spec: { size: 3 }
}
]
EOT

qbec apply default -c crd --yes
qbec apply default -c cr --yes
sed -i 's/size: 3/size: 2/' components/cr.jsonnet
qbec apply default -c cr --yes
```

**Error:**
```
setting cluster to stage
setting context to stage
cluster metadata load took 42ms

will synchronize 1 object(s)

✘ sync memcacheds example-memcached -n test (source cr): the body of the request was in an unknown format - accepted media types include: application/json-patch+json, application/merge-patch+json
command took 260ms
```

**Comment by kvaps** [comment]
Apparently this is connected with the fact that in v1.15 the resources validation was moved out on the server side

**Comment by gotwarlost** [comment]
could you provide the output of `qbec apply -n -c cr default` so I can see what the patch that qbec is trying to apply looks like?

(This works on an older Kubernetes BTW which I think you expect to)

**Comment by gotwarlost** [comment]
never mind I was able to reproduce this problem in minikube. Validation of custom resources now seem to work but by the same token strategic merge patches that are used when this is true do not.

**Comment by gotwarlost** [comment]
fix available in v0.7.5 https://github.com/splunk/qbec/releases/tag/v0.7.5
